filebeat/input/log/harvester.go

```diff
@@ -260,6 +260,16 @@ func (h *Harvester) Run() error {
 		}
 
 		message, err := h.reader.Next()
+
+		// Get copy of state to work on
+		// This is important in case sending is not successful so on shutdown
+		// the old offset is reported
+		state := h.getState()
+
+		// always save state.Offset
+		state.Offset += int64(message.Bytes)
+		h.state = state
```

Comment:
If you move this line here, is there a need for updating it again in line 355?

Reply:
it looks like  unneeded to update h.state again. I'm going to remove it and update PR after testing.